server/pulp/server/async/app.py

```diff
@@ -103,47 +107,26 @@ def get_resource_manager_lock(name):
 
     lock = ResourceManagerLock(name=name)
 
-    with custom_sigterm_handler(name):
-        # Whether this is the first lock availability check for this instance
-        _first_check = True
-
-        while True:
-            # Create / update the worker record so that Pulp knows we exist
-            Worker.objects(name=name).update_one(set__last_heartbeat=datetime.utcnow(),
-                                                 upsert=True)
-            try:
-                lock.save()
-
-                msg = _("Resource manager '%s' has acquired the resource manager lock" % name)
+    # Whether this is the first lock availability check for this instance
+    _first_check = True
+
+    while True:
+        # Create / update the worker record so that Pulp knows we exist
+        Worker.objects(name=name).update_one(set__last_heartbeat=datetime.utcnow(),
+                                             upsert=True)
+        try:
+            lock.save()
+
+            msg = _("Resource manager '%s' has acquired the resource manager lock" % name)
+            _logger.info(msg)
+            break
+        except mongoengine.NotUniqueError:
+            # Only log the message the first time
+            if _first_check:
+                msg = _("Resource manager '%s' attempted to acquire the the resource manager "
```

Comment:
You didn't write this, but this has the same i8ln problem.
Also this needs to be converted to a dictionary based string replacement since this cannot be i8ln to all languages when the positional parameters are reordered.